lumigator/python/mzai/jobs/inference/inference_config.py

```diff
@@ -43,13 +121,13 @@ class SamplingParameters(BaseModel):
 
 
 class PipelineConfig(BaseModel, arbitrary_types_allowed=True):
-    model_path: str = Field(title="The Model HF Hub repo ID", exclude=True)
-    task: str
+    model_path: AssetPath = Field(title="The Model HF Hub repo ID", exclude=True)
+    task: SupportedTask
     revision: str = "main"  # Model version: branch, tag, or commit ID
     use_fast: bool = True  # Whether or not to use a Fast tokenizer if possible
     trust_remote_code: bool = False
-    torch_dtype: str = "auto"
-    accelerator: str = Field(title="Accelerator", default=Accelerator.AUTO, exclude=True)
+    torch_dtype: SerializableTorchDtype = "auto"
```

Comment:
Just a heads up that adding torch as a dependency makes the job much heavier in terms of fingerprint.
As far as I understood, this is required to validate the torch datatype used for e.g. quantization.
Do we need this already for summarization or are we planning to use it later on for larger text generation models?

I am not against it in principle, I just wanted to raise that we might not need this right now (and perhaps find a different solution in the meantime). WDYT?

Reply:
But `torch` is a dependency or `transformers` anyway, how can we move around this?

Reply:
That's right, my bad! Transformers needs to be backed up by torch anyway 👍